lambdas/create_appointment.py

```diff
@@ -1,45 +1,384 @@
+        try:
+            students_query = query(sql,sql_parameters)
+        except Exception as e:
+            raise LambdaException("404: Query to students table for student_id failed: " + str(e))
 
-    query = f"INSERT INTO scheduled_appointments(appointment_id, method, supporter_id, time_of_appt, type, duration, cancelled, cancel_reason, location) VALUES ('{appointment_id}', '{method}', '{supporter_id}', '{time}', '{appt_type}', '{duration}', False, 'None', '{location}');"
-    #INSERT INTO scheduled_appointments(appointment_id, method, supporter_id, time_of_appt, type, duration, cancelled, cancel_reason, location) VALUES ('0', 'test', '2', '01-01-2021', 'Interview', '30', False, 'None', 'CICS');
-    #query = "INSERT INTO scheduled_appointments(appointment_id, method, supporter_id, time_of_appt, type, duration, cancelled, cancel_reason, location) VALUES (" + appointment_id + ", " + method + ", " + supporter_id + ", " + time + ", " + appt_type + ", " + duration
+        if students_query['records'] == []:
+            raise LambdaException("404: Email does not belong to a student.")
+        else:
+            student_id = students_query['records'][0][0]['longValue']
 
-    response = client.execute_statement(
-        database = "postgres",
-        resourceArn = "arn:aws:rds:us-east-2:500514381816:cluster:postgres",
-        secretArn = "arn:aws:secretsmanager:us-east-2:500514381816:secret:rds-db-credentials/cluster-33FXTTBJUA6VTIJBXQWHEGXQRE/postgres-3QyWu7",
-        sql = query,
-    )
+    else:
+        student = int(event['student_id'])
+        # check that student is in DB
+        sql = "SELECT student_id FROM students WHERE student_id = :student"
+        sql_parameters = [
+            {'name' : 'student', 'value': {'longValue': student}}
+        ]
+        # make query
+        try:
+            check_student = query(sql, sql_parameters)
+        except Exception as e:
+            raise LambdaException("404: Query to verify student_id failed: " + str(e))
+        
+        # if student does not exist in DB, raise exception
+        if(check_student['records'] == []):
+            raise LambdaException("404: Student does not exist.")
+        
+        # set student_id variable
+        student_id = student
 
-    # if error, return 404
-    if(response['numberOfRecordsUpdated'] == 0):
-        return {
-            'statusCode': 404, 
-            'body': json.dumps('Error making appointment')
-        }
+    if override and event['supporter_id'] == "":
+        #look up by email
+        if 'supporter_email' not in event:
+            raise LambdaException("404: Must provide supporter_id or email.")
+        # get supporter_email
+        supporter_email = event['supporter_email']
+        # query users table for user_id
+        sql = "SELECT id FROM users WHERE email=:supporter_email;"
+        sql_parameters = [
+            {'name' : 'supporter_email', 'value': {'stringValue' : supporter_email}}
+        ]
```

Comment:
In the input comment, it has supporter_id as an input, yet here it is checking if it is in the event. Just wondering if supporter_id is actually optional or still required.

Reply:
You're right - we added the ability for an admin to look a supporter or student up by email instead of including their id. So either an id or email is required for both student and supporter.